src/WebApp.Web/Features/WeatherForecast/AddWeatherForecast.cs

```diff
@@ -1,39 +1,42 @@
 using FluentValidation;
 using Mapster;
+using Microsoft.EntityFrameworkCore;
 using WebApp.Data;
 
 namespace WebApp.Web.Features.WeatherForecast;
 
-public record AddWeatherForecast(DateOnly Date, int TemperatureC, int WindSpeed, int AirHumidityPercent,
-    int WeatherSummaryId) : IRequest<int>
+public record AddWeatherForecast([Required] DateOnly? Date,
+    [Required, Range(-100, 100)] int? TemperatureC, [Required, Range(0, 50)] int? WindSpeed,
+    [Required, Range(0, 100)] int? AirHumidityPercent,
+    [Required] int? WeatherSummaryId) : IRequest<int>
 {
 }
 
 [UsedImplicitly]
 public class AddWeatherForecastValidator : AbstractValidator<AddWeatherForecast>
 {
-    public AddWeatherForecastValidator()
+    private readonly WebAppDbContext _dbContext;
+
+    public AddWeatherForecastValidator(WebAppDbContext dbContext)
     {
-        RuleFor(x => x.TemperatureC)
-            .NotEmpty()
-            .GreaterThan(100)
-            .LessThan(100);
+        _dbContext = dbContext;
 
-        RuleFor(x => x.AirHumidityPercent)
-            .NotEmpty()
-            .GreaterThan(0)
-            .LessThan(100);
-        
-        RuleFor(x => x.WindSpeed)
-            .NotEmpty()
-            .GreaterThan(0)
-            .LessThan(50);
+        RuleFor(x => x.WeatherSummaryId)
+            .MustAsync(async (weatherSummaryId, ct) =>
+            {
+                return await _dbContext.WeatherSummaries
+                    .AnyAsync(x => x.Id == weatherSummaryId, cancellationToken: ct).ConfigureAwait(false);
```

Comment:
please remove ConfigureAwait. ASP.NET Core doesn't have SynchronizationContext defined. This is redundant.

Reply:
done